x/gov/types/expected_keepers.go

```diff
@@ -48,3 +48,16 @@ type BankKeeper interface {
 	SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error
 	BurnCoins(ctx sdk.Context, name string, amt sdk.Coins) error
 }
+
+// Event Hooks
+// These can be utilized to communicate between a governance keeper and another
+// keepers.
+
+// GovHooks event hooks for governance proposal object (noalias)
+type GovHooks interface {
+	AfterProposalSubmission(ctx sdk.Context, proposalID uint64)                            // Must be called after proposal is submitted
+	AfterProposalDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress) // Must be called after a deposit is made
+	AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress)        // Must be called after a vote on a proposal is cast
+	AfterProposalInactive(ctx sdk.Context, proposalID uint64)                              // Must be called when proposal become inactive
+	AfterProposalActive(ctx sdk.Context, proposalID uint64)                                // Must be called when proposal become active
+}
```

Comment:
not sure if the `AfterProposalSubmission`, `AfterProposalDeposit` and `AfterProposalInactive` hooks are useful?

Reply:
We might not know the uses of these yet, but the point is to expose them in order to make the module more composable.  Instead of needing to get a PR merged everytime someone finds a new hook that's useful for them, it's better to make them all available from the get-go.

But for the sake of discussion, here's an example use case:
- Maybe a DAO community pool wants to deposit matching.  For every x tokens a depositor deposits on a proposal, a DAO wants to match that amount.  That could be done easily using `AfterProposalDeposit`